sdk/src/asset_io.rs

```diff
@@ -48,13 +48,59 @@ impl CAIRead for std::io::Cursor<&mut [u8]> {}
 impl CAIRead for std::io::Cursor<Vec<u8>> {}
 impl CAIRead for NamedTempFile {}
 
+// Helper struct to create a concrete type for CAIRead when
+// that is required
+pub struct CAIReadWrapper<'a> {
```

Comment:
This feels like a workaround and I'm wondering if we can limit its exposure to `pub(crate)` visibility rather than adding it to the API surface.

Reply:
Some APIs only accept concrete instances.  I agree this does not have to be public. This should go away in a future version of Rust.